src/ansi-c/expr2c.cpp

```diff
@@ -3496,6 +3496,22 @@ std::string expr2ct::convert_conditional_target_group(const exprt &src)
   return dest;
 }
 
+std::string expr2ct::convert_bitreverse(const bitreverse_exprt &src)
+{
+  if(auto type_ptr = type_try_dynamic_cast<unsignedbv_typet>(src.type()))
+  {
+    const std::size_t width = type_ptr->get_width();
+    if(width == 8 || width == 16 || width == 32 || width == 64)
+    {
+      return convert_function(
+        src, "__builtin_bitreverse" + std::to_string(width));
+    }
```

Comment:
Belt-and-braces, add an `UNREACHABLE` or an assertion to be more descriptive, instead of using `convert_norep`? Or maybe just have a `PRECONDITION` ?

Reply:
Hmm, there isn't really anything fundamentally wrong in creating a `bitreverse_exprt` with a type that isn't 8/16/32/64 bits wide, we just don't have a good C representation available. So I'd stick with `convert_norep` as that's what the remainder of this code does, if that's ok?

Reply:
Fine by me - it was just a sanity check query from me, thanks.